packages/react/src/search/useSearch.ts

```
@@ -0,0 +1,196 @@
+import { GenericFields, SearchParameters } from '@sitecore-content-sdk/search';
+import { DEFAULT_PAGE, DEFAULT_PAGE_SIZE, getOffset, useSearchService } from './utils';
+
+/**
+ * Options for the useSearch hook.
+ * @public
+ */
+export interface UseSearchOptions<T = GenericFields> {
+  /**
+   * The query string to search for.
+   * By default empty string is used.
+   */
+  query?: string;
+  /**
+   * The current page number (1-indexed).
+   * If not provided, the search will start on page 1.
+   */
+  page?: number;
+  /**
+   * The ID of the search index to use.
+   */
+  searchIndexId: string;
+  /**
+   * Number of results per page.
+   * @default 10
+   */
+  pageSize?: number;
+  /**
+   * Specifies the sorting of the search results.
+   */
+  sort?: SearchParameters<T>['sort'];
+}
+
+/**
+ * The state of the useSearch hook.
+ * @public
+ */
+export interface UseSearchState<T = GenericFields> {
+  /**
+   * The search results.
+   */
+  results: T[];
+  /**
+   * Whether a search request is currently in progress.
+   */
+  isLoading: boolean;
+  /**
+   * Whether the search request was successful.
+   */
+  isSuccess: boolean;
+  /**
+   * Whether the search request failed.
+   */
+  isError: boolean;
+  /**
+   * Total number of results across all pages.
+   */
+  total: number;
+  /**
+   * Total number of pages available based on `total` and `pageSize`.
+   */
+  totalPages: number;
+  /**
+   * The error object if the last search request failed, or null if no error occurred.
+   */
+  error: Error | null;
+}
+
+/**
+ * React hook for performing search queries with pagination.
+ * @param {UseSearchOptions} options - Configuration options for the search hook.
+ * @returns {UseSearchState} The search state.
+ * @public
+ */
+export const useSearch = <T = GenericFields>(
+  options: UseSearchOptions<T>
+): UseSearchState<T> => {
+  const { query, page = DEFAULT_PAGE, searchIndexId, pageSize = DEFAULT_PAGE_SIZE, sort } = options;
+
+  if (!searchIndexId) {
```

Altho is up to the developer to provide the correct configuration, throwing an error here will potentially bring down the whole page. I think we can simply check the value and adapt the initial state of the useState at line 85.

---

It's totally expected, application should fail fast if something is configured in a wrong way, in production you should not have some unexpected issues because it was suppressed by sdk. This is how we approach in content-sdk

---

If the hook is used outside a error boundary (Placeholder) the issue is still there and you can't wrap it in try catch blocks (ofc you can have another abstraction with error boundaries).

---

it should just fail if you develop component and you didn't pass index, why do we need to catch this error and hide?

---

As we discussed, this value can be changed during runtime, so definitely we need to handle it more gracefully